api/app/controllers/spree/api/payments_controller.rb

```diff
@@ -17,8 +17,10 @@ def new
       end
 
       def create
-        @payment = @order.payments.build(payment_params)
-        if @payment.save
+        success = @order.contents.update_cart(payments_attributes: [payment_params])
+        @payment = @order.payments.last
+
+        if success
           respond_with(@payment, status: 201, default_template: :show)
         else
           invalid_resource!(@payment)
```

Comment:
ah cool. if it were me i'd probably want a separate `add_payment` method that update_cart would use if it detects nested params as opposed to the other way around. but i think this is fine as well. :+1: